Support for Mailgun 2.0

Nyholm opened the pull request:
This PR include a few changes: 
* Support for Mailgun 2.0. Which means that we are not longer coupled to Guzzle5. 
* Bugfix where our Extension class extends HttpKernel's extension class instead of the extension class in the DependencyInjection component. 
* Tests on different symfony versions
* Tests for our configuration and extension class
* Dropping support for php 5.3 and php 5.4 as a consequence of updating to Mailgun 2.0. 
* Dropping support for Swiftmailer 4.0

tehplague commented:
Looks good to me. Are there any BC breaks with the current implementation, @Nyholm?

Nyholm commented:
Yes, some. Mailgun is no longer coupled to Guzzle5. Instead we use Httplug. This means that we have to  give a (HttpClient to Mailgun](https://github.com/tehplague/swiftmailer-mailgun-bundle/pull/22/files#diff-04c6e90faac2675aa89e2176d2eec7d8R80) or use Puli and discovery. I've added a [section in the readme](https://github.com/tehplague/swiftmailer-mailgun-bundle/pull/22/files#diff-04c6e90faac2675aa89e2176d2eec7d8R70) to describe this. 

A user that uses our latest version (0.2.1) and upgrades to a version where this PR is merged (0.3.0) will be asked to install a http client since we depend on the virtual package [php-http/client-implementation](http://docs.php-http.org/en/latest/clients.html#composer-virtual-packages). I've added upgrade instructions in Upgrade.md. 

So the BC break is just in the setup/installation not for the actual usage of the Mailgun library.

tehplague commented:
OK, it seems as if we're better off with bumping the major version to 1.0 then. I think the bundle is reasonably stable enough for this.

cedricziel commented:
Any blocker left? If not - how about a merge then? :)

WorldWarIII commented:
Can you merge it to master?

Nyholm commented:
Yes, we should merge this. This PR has been ready for the last 3 months =).
